frontend/src/elm/Pages/Lobby.elm

```diff
@@ -68,18 +70,28 @@ update msg model =
 
         NotificationEvent notificationMsg ->
             let
-                ( notification, cmd1 ) =
-                    Notification.update notificationMsg model.notification
+                ( newSession, notification, cmd1 ) =
+                    Notification.update notificationMsg model.notificationModel model.session
 
                 cmd2 =
                     case notificationMsg of
-                        Notification.OnAcceptFromNotification _ gameID ->
-                            Route.replaceUrl (getNavKey model.session) (Lobby gameID)
+                        Notification.NotificationAccepted _ metadata ->
+                            case metadata of
+                                String gameID ->
+                                    Cmd.batch
+                                        [ LobbyService.closeLobby
+                                        , Route.replaceUrl (getNavKey newSession) (Lobby gameID)
+                                        ]
+
+                        Notification.NotificationDeclined _ metadata ->
+                            case metadata of
+                                String gameID ->
+                                    LobbyService.declineInvitation gameID
 
                         _ ->
                             Cmd.none
             in
-            ( { model | notification = notification }
+            ( { model | notificationModel = notification, session = newSession }
             , Cmd.batch
                 [ Cmd.map NotificationEvent cmd1
```

Comment:
we could move the map operation above, so we would have `Cmd.batch [cmd1, cmd2]` here